### ISSUE TITLE
Batch backward computation for on-policy algorithms

### ISSUE THREAD

**Issue by muupan** [chainer/chainerrl#203]
In ChainerRL's implementation of on-policy algorithms such as A3C, ACER and REINFORCE, forward and backward computation are done like this:
```python
# x2 depends on y1 via a black-box state transition function,
# so these forward computation cannot be batched
y1 = model(x1)
y2 = model(x2)
...
y100 = model(x100)

loss = f(y1, y2, ..., y100)
loss.backward()
```

This is inefficient in Chainer because computation in `loss.backward()` is not batched. To batch backward computation, it is possible to add another batch computation like this:
```
y1 = model(x1)
y2 = model(x2)
...
y100 = model(x100)

ys = model([x1, x2, ..., x100])  # forward computation again, but batched this time
loss = f(ys)
loss.backward()
```
While it introduces redundant recomputation, it may work more efficiently because of batched computation, especially when using GPUs.
